onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlTaggedPointer.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <climits>
+#include <cstdint>
+
+namespace Dml
+{
+
+// D3D12HeapAllocator and D3D12DescriptorHeapAllocator encode the allocation ID
+// into the high bits of the pointers it returns, while the low bits are used as
+// an offset into the allocation. Note that since the layout of bitfields is
+// implementation-defined, you can't just cast a void* into a TaggedPointer: it
+// must be done using masks and shifts.
+struct TaggedPointer
+{
+    static constexpr uint64_t kDeviceIDBits = 4;
+    static constexpr uint64_t kAllocationIDBits = 20;
+    static constexpr uint64_t kOffsetBits = 40;
```

Comment:
```suggestion
    static constexpr uint64_t DeviceIDBits = 4;
    static constexpr uint64_t AllocationIDBits = 20;
    static constexpr uint64_t OffsetBits = 40;
```

[nit] Intra-file/EP consistency. The DML EP doesn't use Hungarian warts like pFoo or kBar (I'm actually really surprised that Google still does adopt that ancient practice).